core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNotificationHelper.java

```diff
@@ -60,6 +61,26 @@ public PolicyNotificationHelper(ServerNotificationHandler notificationHandler, S
     return (Consumer<MessagingException>) e -> fireNotification(e.getEvent(), e, action);
   }
 
+  /**
+   * Creates an exception {@link BiConsumer} that fires a notification using the specified action on success or error.
+   * Notifications are not fired if the {@link BiConsumer} does not receive an {@link CoreEvent} or a {@link MessagingException}.
```

Comment:
the event is not being null-checked, sounds like a contradiction to this javadoc